### PR TITLE
Run verify generated files remake in a tmp directory

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

We already do this in hack/verify-generated-files.sh so we should do it
in verify-generated-files-remake.sh as well.

The idea is that any local changes made because of code generation
should not persist beyond the current run of the script.

Change-Id: I7af176773ae16c393dc2b46c006595243c9fa05b

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/verify-generated-files-remake.sh [hunk 1]
```diff
@@ -21,6 +21,21 @@ set -o pipefail
...
+
+_tmp_gopath="${_tmpdir}/go"
+_tmp_kuberoot="${_tmp_gopath}/src/k8s.io/kubernetes"
+mkdir -p "${_tmp_kuberoot}/.."
+cp -a "${KUBE_ROOT}" "${_tmp_kuberoot}/.."
+
+cd "${_tmp_kuberoot}"
+
+# clean out anything from the temp dir that's not checked in
+git clean -ffxd
```
Comment (BenTheElder) [inline hack/verify-generated-files-remake.sh]: should this just be `-fxd` ?

Comment (dims) [reply hack/verify-generated-files-remake.sh]: @BenTheElder it's a straight cut-n-paste from hack/verify-generated-files.sh, so if it is `-fxd` then we need to fix it there too. just let me know

Comment (dims) [reply hack/verify-generated-files-remake.sh]: looks like "-ffxd" is helpful for cleaning up sub modules which we don't have ... https://stackoverflow.com/questions/9314365/git-clean-is-not-removing-a-submodule-added-to-a-branch-when-switching-branches

**Review by BenTheElder — APPROVED** [review]
/lgtm
thanks dims

**Pull request merged**